frontend/utils/distribution/payments-minimizer.js

```diff
@@ -0,0 +1,9 @@
+'use strict'
+
+// TODO: this algorithm will be responsible for "balancing" payments
+// such that the least number of payments are made.
+function minimizeTotalPaymentsCount (distribution: Array<Object>): Array<any | {|from: string, to: string, amount: number|}> {
```

Comment:
Quick comment: AFAIK the use of `any |` negates anything that follows it, rendering moot any kind of type checking. In other words, AFAIK, `any |` and `any` are the same thing.

Reply:
Ok... you caught me...

I just copied another part of GI's code to fix the flow errors. I forgot where (unfortunately); it might have been Steven's old distribution code.

TBH: I am actually quite unfamiliar with this type of "strict-typing" in JS, that is why I adopted existing code. This fixed flow-errors before pushing the changes (you _completely_ busted me! [perhaps I was trying to publish too fast?]).

I would mad appreciate it if you can help find the correct syntax for removing the first 'any' key-word from this puzzle! Right now, I am way more concerned with the core-distribution algorithm (as I am balancing many perspectives).

For now, I will add solving this syntax puzzle to my TODO list.

Reply:
@taoeffect Update: this is ([after] after) the next item on the TODO list. Going to push cypress fixes, first, and fix a few of the other items, too.

Reply:
I would suggest to create a Payment type so that it can be used to make such function signatures shorter and possibly more useful too:
```js
type Payment {|
  from: string;
  to: string;
  amount: number;
  total: number;
|}

function minimizeTotalPaymentsCount (distribution: Payment[]): Payment[] {
  ...
 ```
 Maybe a Distribution or MonthlyDistribution type could help even further, e.g.:
 ```js
 ...
 type Distribution = Array<Payment>;
 
 function minimizeTotalPaymentsCount (distribution: Distribution): Distribution {
  ...
 ```
 ```

Reply:
Added to local changes, will update PR later.

Thanks for the suggestions. It reads much nicer.